src/Jobs/ErasureJob.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/*
+ * This file is part of blomstra/flarum-gdpr
+ *
+ * Copyright (c) 2021 Blomstra Ltd
+ *
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Blomstra\Gdpr\Jobs;
+
+use Blomstra\Gdpr\Models\ErasureRequest;
+use Flarum\Api\ApiKey;
+use Flarum\Http\AccessToken;
+use Flarum\Queue\AbstractJob;
+use Flarum\User\EmailToken;
+use Flarum\User\LoginProvider;
+use Flarum\User\PasswordToken;
+use Flarum\User\User;
+use Illuminate\Database\ConnectionInterface;
+use Illuminate\Database\Schema\Builder;
+use Illuminate\Support\Str;
+
+class ErasureJob extends AbstractJob
+{
+    /**
+     * @var ErasureRequest
+     */
+    private $erasureRequest;
+
+    /**
+     * @var Builder
+     */
+    protected $schema;
+
+    public function __construct(ErasureRequest $erasureRequest)
+    {
+        $this->erasureRequest = $erasureRequest;
+    }
+
+    public function handle(ConnectionInterface $connection): void
+    {
+        $this->schema = $connection->getSchemaBuilder();
+
+        /** @var User */
+        $user = User::findOrFail($this->erasureRequest->user_id);
```

Comment:
Jobs that throw exceptions will be retried. I assume you want the job to silently fail in case the user no longer exists? If so just `return` or `exit(0)`.